src/main/java/site/coach_coach/coach_coach_server/action/service/ActionService.java

```diff
@@ -1,54 +1,71 @@
 package site.coach_coach.coach_coach_server.action.service;
 
+import java.sql.PreparedStatement;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 import lombok.RequiredArgsConstructor;
 import site.coach_coach.coach_coach_server.action.domain.Action;
-import site.coach_coach.coach_coach_server.action.dto.CreateActionRequest;
-import site.coach_coach.coach_coach_server.action.dto.UpdateActionInfoRequest;
+import site.coach_coach.coach_coach_server.action.dto.ActionDto;
 import site.coach_coach.coach_coach_server.action.repository.ActionRepository;
-import site.coach_coach.coach_coach_server.category.domain.Category;
-import site.coach_coach.coach_coach_server.category.repository.CategoryRepository;
-import site.coach_coach.coach_coach_server.common.constants.ErrorMessage;
-import site.coach_coach.coach_coach_server.common.exception.NotFoundException;
-import site.coach_coach.coach_coach_server.routine.service.RoutineService;
+import site.coach_coach.coach_coach_server.routine.domain.Routine;
 
 @Service
 @RequiredArgsConstructor
 public class ActionService {
-	private final RoutineService routineService;
-	private final CategoryRepository categoryRepository;
+	private static final Logger log = LoggerFactory.getLogger(ActionService.class);
 	private final ActionRepository actionRepository;
+	private final JdbcTemplate jdbcTemplate;
+	private int indexOfAction = 0;
 
 	@Transactional
-	public Long createAction(Long categoryId, Long userIdByJwt,
-		CreateActionRequest createActionRequest) {
-		Category category = categoryRepository.findExistCategory(categoryId)
-			.orElseThrow(() -> new NotFoundException(ErrorMessage.NOT_FOUND_CATEGORY));
+	public void createAction(Routine newRoutine, List<ActionDto> actions) {
```

Comment:
JPA를 사용했을 때 이런 방법도 있을 것 같긴 하네요! 회의 때 방법 아냐고 여쭤보셔서 찾아봤습니당..
```suggestion
	public void createAction(Routine newRoutine, List<ActionDto> actions) {
    List<Action> actionEntities = actions.stream()
        .map(actionDto -> Action.builder()
            .routine(newRoutine)
            .actionName(actionDto.actionName())
            .sets(actionDto.sets())
            .countsOrMinutes(actionDto.countsOrMinutes())
            .build())
        .collect(Collectors.toList());

    // 빈 액션을 채우는 부분
    int emptyActionCount = 4 - actions.size();
    for (int i = 0; i < emptyActionCount; i++) {
        actionEntities.add(Action.builder()
            .routine(newRoutine)
            .actionName(null)
            .sets(null)
            .countsOrMinutes(null)
            .build());
    }
    
    actionRepository.saveAll(actionEntities);
}
```

Reply:
saveAll을 사용한 방법이네요!
저도 찾아봤을 때, JPA 쿼리인 save와 saveAll의 경우엔 단건 삽입을 한다고해서 사용하지않았습니다 ㅠㅠ
한번에 다수의 row를 Insert하는 JPA 방법은 저도 좀 더 찾아보겠습니다! 감사합니다!